ntoskrnl/ke/except.c

```diff
@@ -223,29 +223,37 @@ NtContinue(IN PCONTEXT Context,
     PKTHREAD Thread;
     NTSTATUS Status;
     PKTRAP_FRAME TrapFrame;
+#ifdef _M_IX86
+    PKEXCEPTION_FRAME ExceptionFrame = NULL;
+#else
+    KEXCEPTION_FRAME LocalExceptionFrame;
+    PKEXCEPTION_FRAME ExceptionFrame = &LocalExceptionFrame;
+#endif
 
     /* Get trap frame and link previous one*/
     Thread = KeGetCurrentThread();
     TrapFrame = Thread->TrapFrame;
     Thread->TrapFrame = KiGetLinkedTrapFrame(TrapFrame);
 
     /* Continue from this point on */
-    Status = KiContinue(Context, NULL, TrapFrame);
+    Status = KiContinue(Context, ExceptionFrame, TrapFrame);
     if (NT_SUCCESS(Status))
     {
         /* Check if alert was requested */
         if (TestAlert) KeTestAlertThread(Thread->PreviousMode);
 
         /* Exit to new trap frame */
-        KiServiceExit2(TrapFrame);
+        KiExceptionExit(TrapFrame, ExceptionFrame);
     }
+#ifdef _M_IX86
     else
     {
         /* Exit with an error */
         KiServiceExit(TrapFrame, Status);
```

Comment:
Isn't it possible to have this thing in x64 as well, that does nothing? (This would avoid the 'ifdef' above the 'else')

Reply:
That would just be confusing. We can probably remove the entire thing though.